### ISSUE TITLE
Debian 9 support

### ISSUE THREAD

**Issue by OndrejHome** [OndrejHome/fast-vm#33]
Proof of concept on Debian 9.3:
- `apt-get install libvirt-daemon libvirt-clients libvirt-daemon-system pv ovmf bash-completion dnsmasq-utils make curl sudo thin-provisioning-tools gawk`
- `make install` installation works - will need a DEB package
- need to change `#!/bin/sh` to `#!/bin/bash` in `/usr/bin/fast-vm` for proper operation
- awk seems to match differently the strings here - I have to use `awk  '{ s=substr($0, 0, 5); print s; }'`  on https://github.com/OndrejHome/fast-vm/blob/master/fast-vm#L83  - seems to disappear when `gawk` is installed, need to further check
- for public images we need guestfish command - `apt-get install libguestfs-tools`

Things that needs to be done:
- [x] fast-vm-public-images - debian 9.x image - so we have a stable testing environment 
- [x] instructions in docs/user_guide.txt for Debian - bc342197a8bbe2d1fc0f66a68852d54d432b9fa5
- [x] check why `#!/bin/sh` fails and fix places where needed
- [x] document packages needed on Debian for fast-vm-public-images
- [x] DEB package for Debian

Outlook is optimistic.

**Comment by OndrejHome** [comment]
Proof of concept on Debian 9.3:
- `apt-get install libvirt-daemon libvirt-clients libvirt-daemon-system pv ovmf bash-completion dnsmasq-utils make curl sudo thin-provisioning-tools gawk`
- `make install` installation works - will need a DEB package
- need to change `#!/bin/sh` to `#!/bin/bash` in `/usr/bin/fast-vm` for proper operation
- awk seems to match differently the strings here - I have to use `awk  '{ s=substr($0, 0, 5); print s; }'`  on https://github.com/OndrejHome/fast-vm/blob/master/fast-vm#L83  - seems to disappear when `gawk` is installed, need to further check
- for public images we need guestfish command - `apt-get install libguestfs-tools`

Things that needs to be done:
- [ ] fast-vm-public-images - debian 9.x image - so we have a stable testing environment - already WiP
- [ ] instructions in docs/user_guide.txt for Debian
- [ ] check why `#!/bin/sh` fails and fix places where needed
- [ ] document packages needed on Debian for fast-vm-public-images
- [ ] DEB package for Debian

Outlook is optimistic.

**Comment by OndrejHome** [comment]
- Debian 9.3 image created with preseed, files available in https://github.com/OndrejHome/fast-vm-public-images/commit/db1cdb619878deaf1e124f3c34218a02a70831ec
- image will be available soon after mirror sync at http://ftp.linux.cz/pub/linux/people/ondrej_famera/fastvm-images/generated/6g__debian-9.3.img.xz
- removed bashisms in fast-vm script to allow run debian with dash - not extensively tested - https://github.com/OndrejHome/fast-vm/commit/5f5efa9cddba5a8f4c1035433739e8c042d77281

**Comment by OndrejHome** [comment]
- 02f92a3378658de65e4848e9164df53df4e26cf7 - Testing DEB package: https://www.famera.cz/large_data/files/fast-vm_1.3.1_all.deb
- package needed for public images to work is named `libguestfs-tools` - so installing it should be as simple as `apt-get install libguestfs-tools` - seems that we miss the `selinux-relabel` support here, using a image for libguestfs from other system with support for this works as workaround

**Comment by OndrejHome** [comment]
Installation of fast-vm under Debian:
~~~
# apt-get install gdebi-core
# wget https://www.famera.cz/large_data/files/fast-vm_1.3.1_all.deb
# gdebi fast-vm_1.3.1_all.deb
~~~
`gdebi` is used to resolve and install dependencies properly. In future having a repository might be a better approach so the updates can be delivered.

`libguestfs-tools` can also used the pre-compiled appliance that seems to work properly here with minor changes to system before it can be used (exporting ENV variable).

Basically this is now only missing the documentation in user guide.

**Comment by OndrejHome** [comment]
Ready for 1.4 release.

**Comment by OndrejHome** [comment]
released in 1.4
